app.py

```diff
@@ -6,14 +6,31 @@
 Description: Entry point of the app, contains API implementation.
 """
 
-from fastapi import FastAPI, Header
+from fastapi import FastAPI, Header,File,UploadFile
 from pydantic import BaseModel
 from utils.pinecone_vectorstore import build_pinecone_index
 from utils.chat import get_chat_response
 from auth import is_authorized_request
+import shutil
+from pathlib import Path
+from fastapi.middleware.cors import CORSMiddleware
+from utils.invoice_processing import get_file_type,extract_text_based_on_file_type,fetch_invoice_details
+
 
 app = FastAPI()
 
+# Define the directory to store uploaded files
+UPLOAD_DIR = Path(__file__).resolve().parent / "uploads"
+UPLOAD_DIR.mkdir(exist_ok=True)
+# This example allows all origins, adjust as needed for production
+app.add_middleware(
```

Comment:
What is the use of these middlewares?

Reply:
Was getting cross origin error(405) while sending file to back end that is why used these configuration.